source/components/organisms/Step/Step.js

```diff
@@ -129,7 +131,13 @@ function Step({
               showBackButton={isBackBtnVisible}
               inSubstep={inSubstep}
               onBack={backButtonBehavior}
-              onClose={() => setCloseDialogVisible(true)}
+              onClose={() => {
+                if (inLastMainStep) {
+                  closeForm();
+                } else {
+                  setCloseDialogVisible(true);
+                }
+              }}
```

Comment:
Might be an idea to declare a function for this check (y)

Reply:
Maybe, but I think I disagree. This behaviour is only for this one place; if I define a function for it, you have to jump somewhere else to see what happens. Generally I think small functions that are only used at one place should be avoided, it introduces unneccessary abstraction and clutter. Of course if this ends up reused or grows a bit longer, then it should be broken out into a function, but for now I think I prefer it as it.

Reply:
But you could declare a function in the Step component just as a onClick handler, just a personal preferences to not define functions inside props passing (y)